Error while checking build/O2/fullCI for 14933d8652d31a3c23018b2cf98372162e3ef1bb at 2024-07-10 14:51:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/13288-slc8_x86-64/0/Generators/src/FlowMapper.cxx:30:13: error: use nullptr [modernize-use-nullptr]
/sw/SOURCES/O2/13288-slc8_x86-64/0/Generators/src/FlowMapper.cxx:31:13: error: use nullptr [modernize-use-nullptr]
/sw/SOURCES/O2/13288-slc8_x86-64/0/Generators/src/FlowMapper.cxx:32:10: error: use nullptr [modernize-use-nullptr]
/sw/SOURCES/O2/13288-slc8_x86-64/0/Generators/src/FlowMapper.cxx:63:42: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/13288-slc8_x86-64/0/Generators/src/FlowMapper.cxx:65:43: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/13288-slc8_x86-64/0/Generators/src/FlowMapper.cxx:67:44: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/13288-slc8_x86-64/0/Generators/src/FlowMapper.cxx:105:12: error: use nullptr [modernize-use-nullptr]
/sw/SOURCES/O2/13288-slc8_x86-64/0/Generators/src/FlowMapper.cxx:119:19: error: statement should be inside braces [readability-braces-around-statements]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13288/14933d8652d31a3c23018b2cf98372162e3ef1bb/build_O2_fullCI/pretty.html).

--- a/Generators/include/Generators/FlowMapper.h
+++ b/Generators/include/Generators/FlowMapper.h
@@ -44,24 +44,24 @@ class FlowMapper
  public:
   // Constructor
   FlowMapper();
-  
+
   void Setv2VsPt(TH1D hv2VsPtProvided);
   void SetEccVsB(TH1D hEccVsBProvided);
-  
-  void CreateLUT(); //to be called if all is set
+
+  void CreateLUT(); // to be called if all is set
 
   Double_t MapPhi(Double_t lPhiInput, TH3D* hLUT, Double_t b, Double_t pt);
 
   long binsPhi;             // number of phi bins to use
   double precision = 1e-6;  // could be studied
   double derivative = 1e-4; // could be studied
-  
+
   std::unique_ptr<TH1D> hv2vsPt; // input v2 vs pT from measurement
   std::unique_ptr<TH1D> hEccVsB; // ecc vs B (from Glauber MC or elsewhere)
-  
+
   // Cumulative function to be inverted
   std::unique_ptr<TF1> fCumulative;
-  
+
   // the look-up table
   std::unique_ptr<TH3D> hLUT;
 

--- a/Generators/include/Generators/FlowMapper.h
+++ b/Generators/include/Generators/FlowMapper.h
@@ -44,27 +44,26 @@ class FlowMapper
  public:
   // Constructor
   FlowMapper();
-
-  void Setv2VsPt(TH1D* hv2VsPtProvided);
-  void SetEccVsB(TH1D* hEccVsBProvided);
-
-  void CreateLUT(); // to be called if all is set
-  TH3D* GetLUT();
+  
+  void Setv2VsPt(TH1D hv2VsPtProvided);
+  void SetEccVsB(TH1D hEccVsBProvided);
+  
+  void CreateLUT(); //to be called if all is set
 
   Double_t MapPhi(Double_t lPhiInput, TH3D* hLUT, Double_t b, Double_t pt);
 
   long binsPhi;             // number of phi bins to use
   double precision = 1e-6;  // could be studied
   double derivative = 1e-4; // could be studied
-
-  TH1D* hv2vsPt; // input v2 vs pT from measurement
-  TH1D* hEccVsB; // ecc vs B (from Glauber MC or elsewhere)
-
+  
+  std::unique_ptr<TH1D> hv2vsPt; // input v2 vs pT from measurement
+  std::unique_ptr<TH1D> hEccVsB; // ecc vs B (from Glauber MC or elsewhere)
+  
   // Cumulative function to be inverted
-  TF1* fCumulative;
-
+  std::unique_ptr<TF1> fCumulative;
+  
   // the look-up table
-  TH3D* hLUT;
+  std::unique_ptr<TH3D> hLUT;
 
   ClassDef(FlowMapper, 1);
 };

--- a/Generators/src/FlowMapper.cxx
+++ b/Generators/src/FlowMapper.cxx
@@ -110,8 +110,9 @@ Double_t FlowMapper::MapPhi(Double_t lPhiInput, TH3D* hLUT, Double_t b, Double_t
   // Avoid interpolation problems in dimension: pT
   Double_t lMaxPt = hLUT->GetYaxis()->GetBinCenter(hLUT->GetYaxis()->GetNbins());
   Double_t lMinPt = hLUT->GetYaxis()->GetBinCenter(1);
-  if (pt > lMaxPt)
+  if (pt > lMaxPt){
     pt = lMaxPt; // avoid interpolation problems at edge
+  }
 
   Double_t phiWidth = hLUT->GetZaxis()->GetBinWidth(1); // any bin, assume constant
 

--- a/Generators/src/FlowMapper.cxx
+++ b/Generators/src/FlowMapper.cxx
@@ -26,15 +26,17 @@ namespace eventgen
 FlowMapper::FlowMapper()
 {
   // base constructor. Creates cumulative function only so that's already in place but not much else.
-  fCumulative = std::make_unique<TF1>("fCumulative","x+[0]*TMath::Sin(2*x)", 0,2*TMath::Pi());
+  fCumulative = std::make_unique<TF1>("fCumulative", "x+[0]*TMath::Sin(2*x)", 0, 2 * TMath::Pi());
   binsPhi = 4000; // a first guess
 }
 
-void FlowMapper::Setv2VsPt(TH1D hv2VsPtProvided) {
+void FlowMapper::Setv2VsPt(TH1D hv2VsPtProvided)
+{
   // Sets the v2 vs pT to be used.
   hv2vsPt = std::make_unique<TH1D>(hv2VsPtProvided);
 }
-void FlowMapper::SetEccVsB(TH1D hEccVsBProvided) {
+void FlowMapper::SetEccVsB(TH1D hEccVsBProvided)
+{
   // Sets the v2 vs pT to be used.
   hEccVsB = std::make_unique<TH1D>(hEccVsBProvided);
 }
@@ -52,24 +54,24 @@ void FlowMapper::CreateLUT()
   const Long_t nbinsB = hEccVsB->GetNbinsX();
   const Long_t nbinsPt = hv2vsPt->GetNbinsX();
   const Long_t nbinsPhi = binsPhi; // constant in this context necessary
-  std::vector<double> binsB(nbinsB+1,0);
-  std::vector<double> binsPt(nbinsPt+1,0);
-  std::vector<double> binsPhi(nbinsPhi+1,0);
-  
-  for(int ii=0; ii<nbinsB+1; ii++){
-    binsB[ii] = hEccVsB->GetBinLowEdge(ii+1);
+  std::vector<double> binsB(nbinsB + 1, 0);
+  std::vector<double> binsPt(nbinsPt + 1, 0);
+  std::vector<double> binsPhi(nbinsPhi + 1, 0);
+
+  for (int ii = 0; ii < nbinsB + 1; ii++) {
+    binsB[ii] = hEccVsB->GetBinLowEdge(ii + 1);
   }
-  for(int ii=0; ii<nbinsPt+1; ii++){
-    binsPt[ii] = hv2vsPt->GetBinLowEdge(ii+1);
+  for (int ii = 0; ii < nbinsPt + 1; ii++) {
+    binsPt[ii] = hv2vsPt->GetBinLowEdge(ii + 1);
   }
-  for(int ii=0; ii<nbinsPhi+1; ii++){
-    binsPhi[ii] = static_cast<Double_t>(ii)*2*TMath::Pi()/static_cast<Double_t>(nbinsPhi);
+  for (int ii = 0; ii < nbinsPhi + 1; ii++) {
+    binsPhi[ii] = static_cast<Double_t>(ii) * 2 * TMath::Pi() / static_cast<Double_t>(nbinsPhi);
   }
 
-  //std::make_unique<TH1F>("hSign", "Sign of electric charge;charge sign", 3, -1.5, 1.5);
-  
+  // std::make_unique<TH1F>("hSign", "Sign of electric charge;charge sign", 3, -1.5, 1.5);
+
   hLUT = std::make_unique<TH3D>("hLUT", "", nbinsB, binsB.data(), nbinsPt, binsPt.data(), nbinsPhi, binsPhi.data());
-  
+
   // loop over each centrality (b) bin
   for (int ic = 0; ic < nbinsB; ic++) {
     // loop over each pt bin
@@ -99,9 +101,10 @@ void FlowMapper::CreateLUT()
   }
 }
 
-Double_t FlowMapper::MapPhi(Double_t lPhiInput, TH3D *hLUT, Double_t b, Double_t pt){
-  Int_t lLowestPeriod = TMath::Floor( lPhiInput/(2*TMath::Pi()) );
-  Double_t lPhiOld = lPhiInput - 2*lLowestPeriod*TMath::Pi();
+Double_t FlowMapper::MapPhi(Double_t lPhiInput, TH3D* hLUT, Double_t b, Double_t pt)
+{
+  Int_t lLowestPeriod = TMath::Floor(lPhiInput / (2 * TMath::Pi()));
+  Double_t lPhiOld = lPhiInput - 2 * lLowestPeriod * TMath::Pi();
   Double_t lPhiNew = lPhiOld;
 
   // Avoid interpolation problems in dimension: pT

--- a/Generators/src/FlowMapper.cxx
+++ b/Generators/src/FlowMapper.cxx
@@ -26,23 +26,17 @@ namespace eventgen
 FlowMapper::FlowMapper()
 {
   // base constructor. Creates cumulative function only so that's already in place but not much else.
-  fCumulative = new TF1("fCumulative", "x+[0]*TMath::Sin(2*x)", 0, 2 * TMath::Pi());
-  hv2vsPt = 0x0;
-  hEccVsB = 0x0;
-  hLUT = 0x0;
-
+  fCumulative = std::make_unique<TF1>("fCumulative","x+[0]*TMath::Sin(2*x)", 0,2*TMath::Pi());
   binsPhi = 4000; // a first guess
 }
 
-void FlowMapper::Setv2VsPt(TH1D* hv2VsPtProvided)
-{
+void FlowMapper::Setv2VsPt(TH1D hv2VsPtProvided) {
   // Sets the v2 vs pT to be used.
-  hv2vsPt = hv2VsPtProvided;
+  hv2vsPt = std::make_unique<TH1D>(hv2VsPtProvided);
 }
-void FlowMapper::SetEccVsB(TH1D* hEccVsBProvided)
-{
+void FlowMapper::SetEccVsB(TH1D hEccVsBProvided) {
   // Sets the v2 vs pT to be used.
-  hEccVsB = hEccVsBProvided;
+  hEccVsB = std::make_unique<TH1D>(hEccVsBProvided);
 }
 void FlowMapper::CreateLUT()
 {
@@ -58,17 +52,24 @@ void FlowMapper::CreateLUT()
   const Long_t nbinsB = hEccVsB->GetNbinsX();
   const Long_t nbinsPt = hv2vsPt->GetNbinsX();
   const Long_t nbinsPhi = binsPhi; // constant in this context necessary
-  Double_t binsB[nbinsB + 1], binsPt[nbinsPt + 1], binsPhi[nbinsPhi + 1];
+  std::vector<double> binsB(nbinsB+1,0);
+  std::vector<double> binsPt(nbinsPt+1,0);
+  std::vector<double> binsPhi(nbinsPhi+1,0);
+  
+  for(int ii=0; ii<nbinsB+1; ii++){
+    binsB[ii] = hEccVsB->GetBinLowEdge(ii+1);
+  }
+  for(int ii=0; ii<nbinsPt+1; ii++){
+    binsPt[ii] = hv2vsPt->GetBinLowEdge(ii+1);
+  }
+  for(int ii=0; ii<nbinsPhi+1; ii++){
+    binsPhi[ii] = static_cast<Double_t>(ii)*2*TMath::Pi()/static_cast<Double_t>(nbinsPhi);
+  }
 
-  for (int ii = 0; ii < nbinsB + 1; ii++)
-    binsB[ii] = hEccVsB->GetBinLowEdge(ii + 1);
-  for (int ii = 0; ii < nbinsPt + 1; ii++)
-    binsPt[ii] = hv2vsPt->GetBinLowEdge(ii + 1);
-  for (int ii = 0; ii < nbinsPhi + 1; ii++)
-    binsPhi[ii] = static_cast<Double_t>(ii) * 2 * TMath::Pi() / static_cast<Double_t>(nbinsPhi);
-
-  hLUT = new TH3D("hLUT", "", nbinsB, binsB, nbinsPt, binsPt, nbinsPhi, binsPhi);
-
+  //std::make_unique<TH1F>("hSign", "Sign of electric charge;charge sign", 3, -1.5, 1.5);
+  
+  hLUT = std::make_unique<TH3D>("hLUT", "", nbinsB, binsB.data(), nbinsPt, binsPt.data(), nbinsPhi, binsPhi.data());
+  
   // loop over each centrality (b) bin
   for (int ic = 0; ic < nbinsB; ic++) {
     // loop over each pt bin
@@ -98,19 +99,9 @@ void FlowMapper::CreateLUT()
   }
 }
 
-TH3D* FlowMapper::GetLUT()
-{
-  if (!hLUT) {
-    LOG(info) << "LUT was not created! Returning null pointer, beware";
-    return 0x0;
-  }
-  return hLUT;
-}
-
-Double_t FlowMapper::MapPhi(Double_t lPhiInput, TH3D* hLUT, Double_t b, Double_t pt)
-{
-  Int_t lLowestPeriod = TMath::Floor(lPhiInput / (2 * TMath::Pi()));
-  Double_t lPhiOld = lPhiInput - 2 * lLowestPeriod * TMath::Pi();
+Double_t FlowMapper::MapPhi(Double_t lPhiInput, TH3D *hLUT, Double_t b, Double_t pt){
+  Int_t lLowestPeriod = TMath::Floor( lPhiInput/(2*TMath::Pi()) );
+  Double_t lPhiOld = lPhiInput - 2*lLowestPeriod*TMath::Pi();
   Double_t lPhiNew = lPhiOld;
 
   // Avoid interpolation problems in dimension: pT

